### ISSUE TITLE
How to honour translucency setting?

### ISSUE THREAD

**Issue by claremacrae** [damiankorcz/Prism-Theme#30]
Hi,

This is not urgent!! But since I spent some time on it, I thought I would at least write it up whilst it's in my mind...

**Obsidian Settings**

I have Translucent Window turned off in the main Appearance settings:

![image](https://user-images.githubusercontent.com/4840096/151620324-0e963657-f2b4-4702-8f54-4ad03687bebc.png)

**Prism behaviour on my Mac**

With Prism, there are some elements of the Obsidian UI that are still drawn non-opaque.

Depending on what is in the Window behind, it can be quite hard to read the text on the modal in front...

**Example screenshots**

Command prompt:

![image](https://user-images.githubusercontent.com/4840096/151620737-367a6b56-9e0f-41d8-a703-5947e3a30fe7.png)

Plugin startup times:

![image](https://user-images.githubusercontent.com/4840096/151620770-b93c3315-945a-4107-94ec-92882f84ab9c.png)

It's not everything though: The Settings dialog is fully opaque.

**Things I tried**

I've been through all the settings in Style Settings, and also read through all the sections of the User Interface bit of the Wiki...

So it seems like there isn't a way in the Style Settings to control this.

**Suggestions**

1. Ideally, could Prism honour the Obsidian settings and if translucent is turned off, then don't have anything translucent in Prism
2. If that can't be done, could a toggle be added for Prism, to totally turn of translucency please?

Many thanks, as always, for a really nice theme!

**Comment by damiankorcz** [comment]
Hi @claremacrae,

I've specifically made them have the frosted translucent effect mostly because I like the effect and personally didn't have trouble with readability; although I did adjust it to be less "frosted" in a recent update so that might have made it worse.

I can totally see why you might not like it though. I'll look into adding a toggle to disable it and have an opaque background instead. Not sure if I can tie this specifically to the Translucent Window toggle since that impacts the entire window and I haven't really looked into how that specifically works in Obsidian yet.

P.S. Feel free to @ me in Discord if you don't want to spend the time figuring out a setting + writing up an issue (I do appreciate it a lot though). I'm pretty active there so I will likely get back to you promptly.

**Comment by claremacrae** [comment]
Hi @damiankorcz 

That's great - thanks.

I actually don't mind writing up issues at all... I kind of feel that when I'm using a project that someone else has volunteered their time on, it's incumbent on me to work through an issue carefully enough to be confident in what I'm asking for.

The other benefit is that there is then a write-up of the reasoning for the issue in the project, so if another user comes along and is asking a similar thing, they can benefit from the write-up (like, if a workaround were suggested in the issue, for example..)

The other thing is that links to issues I've logged have turned out to be useful reminders to me in my own Obsidian vault...  and the screenshots take up GitHub diskspace instead of my own Obsidian Sync space! :-)

**Comment by claremacrae** [comment]
> I can totally see why you might not like it though. I'll look into adding a toggle to disable it and have an opaque background instead. Not sure if I can tie this specifically to the Translucent Window toggle since that impacts the entire window and I haven't really looked into how that specifically works in Obsidian yet.

I was more asking if you could use the Obsidian toggle to determine the default behaviour in Prism... If user has Obsidian translucency turned off, then turn off Prism translucency by default...

**Comment by claremacrae** [comment]
Sorry for the nonsense initial summary... Now improved.

**Comment by claremacrae** [comment]
Is there a snippet I could add, as a workaround for this?

**Comment by damiankorcz** [comment]
> Is there a snippet I could add, as a workaround for this?

You can do this for now as a snippet to make the background opaque:

```CSS
.theme-light {
--background-secondary-blur: hsl(220, 13%, 95%, 1);
}

.theme-dark {
--background-secondary-blur: hsla(0, 0%, 15%, 1);
}
```

**Comment by claremacrae** [comment]
Thank you - that worked a treat!

**Comment by damiankorcz** [comment]
This should now be addressed in Version 1.5.0. I've adjusted the Frosted Blur background effect to improve text readability. (Can be made opaque in Style Settings under Feature Toggles). Also you shouldn't need the snippet for changing the note background. I've gone back to a pure white background in this version.

**Comment by claremacrae** [comment]
> This should now be addressed in Version 1.5.0. I've adjusted the Frosted Blur background effect to improve text readability. (Can be made opaque in Style Settings under Feature Toggles). Also you shouldn't need the snippet for changing the note background. I've gone back to a pure white background in this version.

Fabulous - thank you very much! (I settled on a lighter gray background than the them was using, so I'll happily stick with that - it was helpful)
